sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecutionSuite.scala

```diff
@@ -68,4 +84,74 @@ class MicroBatchExecutionSuite extends StreamTest with BeforeAndAfter {
       CheckNewAnswer((25, 1), (30, 1))   // This should not throw the error reported in SPARK-24156
     )
   }
+
+  test("SPARK-30143: stop waits until timeout if blocked") {
```

Comment:
I'm not sure that's an over-engineering, but it seems to repeat same test with different trigger, which makes me think it might be ideal to have set of tests which can apply both micro-batch and continuous mode and move them to the base class of suites.